toolkit/tools/pkg/imagecustomizerlib/artifactsinputoutput.go

```diff
@@ -111,3 +157,157 @@ func outputArtifacts(items []imagecustomizerapi.OutputArtifactsItemType, outputD
 
 	return nil
 }
+
+func replaceSuffix(input string, oldSuffix string, newSuffix string) string {
+	if !strings.HasSuffix(input, oldSuffix) {
+		return input
+	}
+	return strings.TrimSuffix(input, oldSuffix) + newSuffix
+}
+
+func writeInjectFilesYaml(metadata []imagecustomizerapi.InjectArtifactMetadata, outputDir string) error {
+	yamlStruct := imagecustomizerapi.InjectFilesYaml{
+		InjectFiles: metadata,
+	}
+
+	// Marshal YAML
+	yamlBytes, err := yaml.Marshal(&yamlStruct)
+	if err != nil {
+		return fmt.Errorf("failed to marshal inject files metadata: %w", err)
+	}
+
+	// Prepend header comment
+	var builder strings.Builder
+	builder.WriteString("# This file is generated automatically by Prism output artifacts API.\n\n")
```

Comment:
I don't know if this message is needed. I think there are some scenarios where we expect the user to modify this file. For example, to add IPE policies. So, we don't need to warn the user that this file was auto-generated.